src/software/thunderscope/widget_setup_functions.py

```diff
@@ -0,0 +1,283 @@
+    )
+
+    for arg in [
+        (World, world.world_buffer),
+        (RobotStatus, world.robot_status_buffer),
+        (Referee, world.referee_buffer),
+        (PrimitiveSet, obstacles.primitive_set_buffer),
+        (PrimitiveSet, paths.primitive_set_buffer),
+        (PassVisualization, passing.pass_visualization_buffer),
+        (ValidationProtoSet, validation.validation_set_buffer),
+        (SimulatorState, sim_state.simulator_state_buffer),
+    ] + [
+        (HRVOVisualization, hrvo_sim_state.hrvo_buffer)
+        for hrvo_sim_state in hrvo_sim_states
+    ]:
+        full_system_proto_unix_io.register_observer(*arg)
+
+    return field
+
+
+def setup_parameter_widget(proto_unix_io, friendly_colour_yellow):
+    """Setup the parameter widget
+
+    :param proto_unix_io: The proto unix io object
+    :param friendly_colour_yellow:
+    :returns: The proto configuration widget
+
+    """
+
+    config = ThunderbotsConfig()
+    config.sensor_fusion_config.friendly_color_yellow = friendly_colour_yellow
+
+    def on_change_callback(attr, value, updated_proto):
+        proto_unix_io.send_proto(ThunderbotsConfig, updated_proto)
+
+    return ProtoConfigurationWidget(config, on_change_callback)
+
+
+def setup_log_widget(proto_unix_io):
+    """Setup the wiget that receives logs from full system
+
+    :param proto_unix_io: The proto unix io object
+    :returns: The log widget
+
+    """
+    # Create widget
+    logs = g3logWidget()
+
+    # Register observer
+    proto_unix_io.register_observer(RobotLog, logs.log_buffer)
+
+    return logs
+
+
+def setup_performance_plot(proto_unix_io):
+    """Setup the performance plot
+
+    :param proto_unix_io: The proto unix io object
+    :returns: The performance plot widget
+
+    """
+
+    def extract_namedvalue_data(named_value_data):
+        return {named_value_data.name: named_value_data.value}
+
+    # Performance Plots plot HZ so the values can't be negative
+    proto_plotter = ProtoPlotter(
+        min_y=0,
+        max_y=100,
+        window_secs=15,
+        configuration={NamedValue: extract_namedvalue_data},
+    )
+
+    # Register observer
+    proto_unix_io.register_observer(NamedValue, proto_plotter.buffers[NamedValue])
+    return proto_plotter
+
+
+def setup_play_info(proto_unix_io):
+    """Setup the play info widget
+
+    :param proto_unix_io: The proto unix io object
+    :returns: The play info widget
+
+    """
+
+    play_info = PlayInfoWidget()
+    proto_unix_io.register_observer(PlayInfo, play_info.playinfo_buffer)
+    return play_info
+
+
+def setup_referee_info(proto_unix_io):
+    """Setup the referee info widget
+
+    :param proto_unix_io: The proto unix io object
+    :returns: The referee info widget
+
+    """
+
+    referee_info = RefereeInfoWidget()
+    proto_unix_io.register_observer(Referee, referee_info.referee_buffer)
+
+    return referee_info
+
+
+def setup_cost_visualization_widget(self, proto_unix_io):
```

Comment:
There's a bug here with launching the cost vis widget. Note that the `self` var is used below as well
```suggestion
def setup_cost_visualization_widget(proto_unix_io):
```